cors/mod.ts

```diff
@@ -0,0 +1,207 @@
+    // The request is outside the scope of this specification.
+    const requestOrigin = request.headers.get("Origin");
+
+    // Always set Vary header
+    // https://github.com/rs/cors/issues/10
+    vary(
+      (header: string) => response.headers.get(header) || "",
+      (header: string, value: string) => {
+        response.headers.set(header, value);
+      },
+      "origin",
+    );
+    if (!requestOrigin) {
+      return;
+    }
+
+    let o;
+
+    if (!options.origin || options.origin === "*") {
+      // allow any origin
+      o = "*";
+    } else if (typeof options.origin === "string") {
+      // fixed origin
+      o = options.origin;
+    } else if (typeof options.origin === "function") {
+      o = options.origin(request, response);
+
+      if (o instanceof Promise) {
+        o = await o;
+      }
+
+      if (!o) {
+        // Safari (and potentially other browsers) need content-length 0,
+        //   for 204 or they just hang waiting for a body
+        response.status_code = options.optionsSuccessStatus as number;
+        response.headers.set("Content-Length", "0");
+
+        return;
+      }
+    } else {
+      // reflect origin
+      o = isOriginAllowed(requestOrigin, options.origin) ? requestOrigin
+      : false;
+    }
+
+    let c;
+
+    if (typeof options.credentials === "function") {
+      c = options.credentials(request, response);
+
+      if (c instanceof Promise) {
+        c = await c;
+      }
+    } else {
+      c = !!options.credentials;
+    }
+
+    const headersSet: { [key: string]: string } = {};
+
+    function set(key: string, value: string) {
+      response.headers.set(key, value);
+      headersSet[key] = value;
+    }
+
+    // If there is no Access-Control-Request-Method header or if parsing failed,
+    // do not set any additional headers and terminate this set of steps.
+    // The request is outside the scope of this specification.
+    if (
+      request.method !== "OPTIONS" ||
+      !request.headers.get("Access-Control-Request-Method")
+    ) {
+      // Simple Cross-Origin Request, Actual Request, and Redirects
+      set("Access-Control-Allow-Origin", o);
+
+      if (c === true) {
+        set("Access-Control-Allow-Credentials", "true");
+      }
+
+      if (typeof options.exposeHeaders === "string") {
+        set("Access-Control-Expose-Headers", options.exposeHeaders);
+      }
+    } else if (options.preflight) {
+      // Preflight Request
+      response.headers.set("Access-Control-Allow-Origin", o);
+
+      if (c === true) {
+        response.headers.set("Access-Control-Allow-Credentials", "true");
+      }
+
+      if (options.maxAge) {
+        response.headers.set("Access-Control-Max-Age", options.maxAge);
+      }
+      if (typeof options.allowMethods === "string") {
+        response.headers.set(
+          "Access-Control-Allow-Methods",
+          options.allowMethods,
+        );
+      }
+
+      let aheaders = options.allowHeaders;
```

Comment:
Could this be typed to be `allowHeaders`?

Reply:
https://github.com/drashland/deno-drash-middleware/pull/1/commits/1627c064bf674eeeccbe726cd79bd685da78dd8b